frontend/src/services/transactions/request-manager.ts

```diff
@@ -104,23 +104,45 @@ export async function getRequestData(
     throw new Error('No request known for this identifier!');
   }
 }
-
-export async function checkIfRequestHasExpired(
+type RequestExpiryInfo = {
+  validityExpired: boolean;
+  noActiveClaims: boolean;
+  notWithdrawnBySomeoneElse: boolean;
+  timeToExpiredMillis: number;
+};
+export async function getRequestExpiryInfo(
   rpcUrl: string,
   requestManagerAddress: string,
   requestIdentifier: UInt256,
   requestAccount: string,
-): Promise<boolean> {
-  const request = await getRequestData(rpcUrl, requestManagerAddress, requestIdentifier);
+): Promise<RequestExpiryInfo> {
+  const { validUntil, activeClaims, withdrawInfo } = await getRequestData(
+    rpcUrl,
+    requestManagerAddress,
+    requestIdentifier,
+  );
   const provider = new JsonRpcProvider(rpcUrl);
   const block = await provider.getBlock('latest');
-  const validityExpired = request.validUntil.lt(block.timestamp);
-  const noActiveClaims = request.activeClaims.eq(0);
+  const validityExpired = validUntil.lt(block.timestamp);
+  const noActiveClaims = activeClaims.eq(0);
   const notWithdrawnBySomeoneElse =
-    request.withdrawInfo.filler.toLowerCase() == '0x0000000000000000000000000000000000000000' ||
-    request.withdrawInfo.filler.toLowerCase() == requestAccount.toLowerCase();
+    withdrawInfo.filler.toLowerCase() == '0x0000000000000000000000000000000000000000' ||
+    withdrawInfo.filler.toLowerCase() == requestAccount.toLowerCase();
 
-  return validityExpired && noActiveClaims && notWithdrawnBySomeoneElse;
+  const timeNow = BigNumber.from(
+    Math.ceil(Date.now() / 1000), // to seconds
+  );
+  const timeToExpiredMillis = Math.max(
+    validUntil.sub(timeNow).toNumber() * 1000, // to milliseconds
```

Comment:
Why not convert the `validUntil` to milliseconds before the call of `sub` instead of having two conversions here?

Reply:
<img src="https://media0.giphy.com/media/AjYsTtVxEEBPO/giphy.gif"/>
Pretty new to the operations with BigNumbers, don't mind this rookie mistake. Thanks a lot for pointing out!